### PR TITLE
hw/sensor: fix possible null dereference in sensor_set_thresh()

### PR DESCRIPTION
**Pull request by KKopyscinski**
Added assert to check if stt is not NULL before dereferencing it

### CODE REVIEW

File: hw/sensor/src/sensor.c [hunk 1]
```diff
@@ -1970,14 +1970,15 @@ sensor_set_thresh(const char *devname, struct sensor_type_traits *stt)
     struct sensor *sensor;
     int rc;
 
+    assert(stt != NULL);
```
Comment (andrzej-kaczmarek) [inline hw/sensor/src/sensor.c:1973]: you should better return error here

Comment (KKopyscinski) [reply hw/sensor/src/sensor.c]: done

File: hw/sensor/src/sensor.c [hunk 2]
```diff
@@ -1970,14 +1970,19 @@ sensor_set_thresh(const char *devname, struct sensor_type_traits *stt)
     struct sensor *sensor;
     int rc;
 
+    if (stt != NULL) {
```
Comment (andrzej-kaczmarek) [inline hw/sensor/src/sensor.c:1973]: so now I am confused... we want to avoid null dereference or force it? :D

also update commit message

Comment (KKopyscinski) [reply hw/sensor/src/sensor.c]: avoid null, of course... :)
updated

**Review by andrzej-kaczmarek — APPROVED** [review]

**Pull request merged**